docs/security.md

```diff
@@ -0,0 +1,48 @@
+
+Sometimes, it is necessary to have solution values that logistically cannot be signed, but also should not be changed.  In scenarios like these, you should try to have a signed coin use announcements to assert that the coin is being spent with the correct information.
+
+## Asserting solution truth
+
+Signing is how you prevent nodes from messing with your own spends, but sometimes you want to create coins that will be traded around with specific rules.  As a result you don't know who will be spending the coin, and you don't know if they will be honest.  We saw in [our discussion of outer puzzles](https://chialisp.com/docs/common_functions#outer-and-inner-puzzles) that you can enforce rules on your child coins using currying and wrapping tree hashes, but there are times when you also want to enforce truths about yourself or your parent.
+
+This is where the `ASSERT_MY_*` family of opcodes comes in.  When you need information (`parent_coin_info`, `puzzle_hash`, `amount`) about your coin to use in the puzzle, it cannot always be curried in by an honest party.  Sometimes, it will need to be passed in through the solution.  The solution should always be treated as if it is being solved by malicious or careless parties.  If any coin information is being passed in, it should be asserted with opcodes to ensure that the network, who can see that information, can confirm it.
+
+Keep in mind that `ASSERT_MY_COIN_ID` will actually implicitly assert all three of the pieces of information in a coin. The same is true of `ASSERT_MY_PARENT_ID` for parent coins, which is particularly useful since there is no such thing as `ASSERT_MY_PARENT_PUZZLE_HASH`, for example.
+
+## Replay attacks
+
+Another huge concern when creating your spends is whether they will be valid if parts of them are excluded or reused.  This kind of attack is the reason why `AGG_SIG_UNSAFE` is labeled the way it is.
+
+If you sign something with `AGG_SIG_UNSAFE`, the only data that is being signed is the message you are trying to sign.  Once you sign and push it, that signature lives on the blockchain forever.  If you later create a puzzle that is locked up with the need for the same signature, an attacker can find the signature you used last time and reuse it.  This is why you should try to always use `AGG_SIG_ME` if possible.  Not only does it require that you also sign the coin ID (something that should be unique to every spend), but it also requires that you sign the genesis challenge of the network you are on. Signatures of the same coins on testnet can't possibly work on mainnet.
```

Comment:
To be precise, the genesis challenge and coin id are *committed to* in the thing that's signed... it's not exactly true that you sign the genesis challenge, so this text might be slightly misleading.